keps/sig-api-machinery/20190802-dynamic-coordinated-storage-version.md

```diff
@@ -172,6 +172,35 @@ During bootstrap, for each resource, the API server
   to avoid conflicting with other API servers.
 * installs the resource handler.
 
+The above mentioned process requires an API server to update the storageVersion
+before accepting API requests. If we don't enforce this order, data encoded in
+an unexpected version can sneak into etcd. For example, an API server persists a
+write request encoded in an obsoleted version, then it crashes before it can
+update the storageVersion. The storage migrator has no way to detect this write.
+
+For the cmd/kube-apiserver binary, we plan to enforce this order by adding a new
+filter to the [handler chain][]. Before kube-aggregator, kube-apiserver, and
+apiextension-apiserver have registered the storage version of the built-in
+resources they host, this filter only allows the following requests to pass:
+1. a request sent by the loopbackClient and is destined to the storageVersion
+   API.
+2. the verb of the request is GET.
+3. the request is for an API that is not persisted, e.g.,
+   SubjectAccessReview and TokenReview. [Here] is a complete list.
+4. the request is for an aggregated API or an API defined by CRD.
+
+The filter rejects other requests with a 404 response code.
```

Comment:
A 404 here may cause controllers or other clients to do weird things.  Is there a "not ready yet" return code?  A 429 may work in a pinch.

Reply:
Yeah, a 503, that's what I was looking for.